### DISCUSSION TITLE
Help to add 7z packing capability

### DISCUSSION THREAD

**Discussion by zEdS15B3GCwq** [coderaiser/cloudcmd#345] [Q&A]
Hi,

Seeing how helpful you were in my other discussion thread, I'm asking for your advice regarding one more matter. Instead of the built-in tar or zip support, I'd like to use 7-zip to pack files; more specifically, I want to be able to select a number of files and pack them into multi-volume 7z archives, e.g. `7z a -v200M -mx0 archivename @filelist`.

The Alpine container only has a package for 7z (p7zip), so I'm discussing 7z; RAR would be just as good.

My JS skill is elementary, so correct me if I'm wrong in my assumptions below.

I've looked a bit into the User Menu, which, I assume, runs client-side JS. The way I understand things, I could use `CloudCmd.TerminalRun.show()` with the 7z command, for which I believe I'd need to install a terminal with CC.

However, for reasons, I prefer to restrict CC to no console and no terminal. Is there any other way to run a server-side script, for example using `child_process.exec()`, besides extending the server-side source code?

I've peeked into https://github.com/cloudcmd/cloudcmd-plugin, but that mentions loading a client-side script, so I'm assuming that's not much different from what I can do with the User Menu.

Any hints how you'd solve this?

**Comment by coderaiser** [comment]
`cloudcmd-plugin` is for client side only, but if you don’t want to use `terminal` (what is very sad, because it feet’s very good to run any cli tool from `User Menu`), you can up your own web server which will run `7z` for you, and run in from `cloudcmd-plugin` with help of `http`-request.

**Reply by coderaiser** [reply]
Hold up, support of `cloudcmd-plugin` was [dropped a couple versions ago](https://github.com/coderaiser/cloudcmd/commit/a61eb22b62dcdcb519133d1dd4c2931bb00c4649) in favor of `User Menu`. Anyways you still can send Ajax-request from it.
